xdsl/dialects/bigint.py

```diff
@@ -1,7 +1,25 @@
 """Dialect for arbitrary-precision integers."""
```

Comment:
So it turns out that LLVM's APInt or arbitrary-precision integer is different in that it holds both the value and the bitwidth (so you can have an infinite number of representations of the value 0, one for each bitwidth, if I understand correctly). In that way, this dialect is not for arbitrary-precision integers, but specifically as a way to model methods of Python's int class. This is especially the case if we add a method per `<`, `<=`, `>`, `>=`, which are otherwise redundant. It feels worth expanding on the description here to explain all this.

Reply:
Regarding `APInt` that is correct, any equality checks the bit width value first.

Regarding this PR, I think more documentation would benefit those that are not tagged in these PRs or don't fully understand the utility of this/where this is going. IMHO, more input the better it is, even if this can be provided sporadically.